File: src/main/java/com/example/auction/model/UserAccount.java
package com.example.auction.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import javax.persistence.*;
import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class UserAccount {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    //@Size(min = 5, message = "Name must be at least 5 characters long")
    private String accountName;

    private String email;

    private String password;

    @OneToOne
    private Address address;

    @Temporal(TemporalType.DATE)
    private Date accountCreated;

    @Enumerated(EnumType.STRING)
    private AccountStatus accountStatus;

    @Enumerated(EnumType.STRING)
    private AccountType accountType;

    private boolean active;


}

File: angular-client/src/app/models/useraccount.ts
import {AccountStatus} from "./accountstatus";
import {AccountType} from "./account-type";
import {Address} from "./address";

export class UserAccount {
  constructor(private id: number,
              private accountName: string,
              private email: string,
              private password: string,
              private address: Address,
              private accountCreated: Date,
              private accountStatus: AccountStatus,
              private accountType: AccountType,
              private active: boolean) {
  }
}
